core-parser/src/main/java/org/uniprot/core/parser/fasta/uniprot/UniProtKBFastaParserWriter.java

```diff
@@ -24,15 +24,19 @@ static String toString(UniProtKBFasta entry) {
         sb.append(entry.getId());
         sb.append('|');
         sb.append(entry.getUniProtkbId().getValue());
-        sb.append(' ').append(getProteinName(entry));
-        sb.append(" OS=").append(entry.getOrganism().getScientificName());
-        sb.append(" OX=").append(entry.getOrganism().getTaxonId());
+        if (Utils.notNullNotEmpty(entry.getProteinName())) {
+            sb.append(' ').append(getProteinName(entry));
+        }
+        if (Utils.notNull(entry.getOrganism())) {
+            sb.append(" OS=").append(entry.getOrganism().getScientificName());
+            sb.append(" OX=").append(entry.getOrganism().getTaxonId());
+        }
 
         if (Utils.notNullNotEmpty(entry.getGeneName())) {
             sb.append(" GN=").append(entry.getGeneName());
         }
 
-        if (isNotSwissProtIsoform(entry)) {
+        if (isNotSwissProtIsoform(entry) && Utils.notNull(entry.getSequenceVersion())) {
```

Comment:
write seperate test - 1. for isoform PE and SV not there 2. seqversion null will not have PE and SV and remaining 2 cases.

In short write all four cases tests expilictly. 
#this is seems a busniess rule - PE and SV only present for swissprot entries who has sequence version number.

I think this rule should be explicit with all tests